oi/arch/Arch.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+struct user_regs_struct;
+
+namespace oi::detail::arch {
+
+void setProgramCounter(user_regs_struct& regs, uintptr_t pc);
+
+std::optional<uintptr_t> getReturnValueAddress(const user_regs_struct&);
```

Comment:
Why return an optional?

Reply:
I mainly did this to match the existing API, though it was never optional there either. This way it might be possible to part implement an architecture if it's ever difficult to get the return value/guess an arg position in future.